### PR TITLE
fix daikon for java 17, now only miss javascript engine

### PR DESCRIPTION
**Pull request by wwang-talend**
**What is the problem this Pull Request is trying to solve?**
https://jira.talendforge.org/browse/TDI-46665
 
**What is the chosen solution to this problem?**
https://jira.talendforge.org/browse/TDI-46665

 
**Please check if the Pull Request fulfills these requirements**
- [x] The PR commit message follows our [guidelines](https://github.com/Talend/daikon/blob/master/CONTRIBUTING.md)
- [ ] Tests for the changes have been added (for bug fixes / features, coverage should be over 75% in the new code)
- [ ] Docs have been added / updated (for bug fixes / features)
- [ ] Related design / discussions / pages (not in JIRA), if any, are all linked or available in the Pull Request

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -203,7 +203,7 @@
                 <groupId>com.cedarsoftware</groupId>
                 <artifactId>json-io</artifactId>
                 <!-- using a specific Talend artifact version until the next official release, the PR has been merge into master but not yet release. -->
-                <version>4.9.9-TALEND</version>
+                <version>4.13.0</version>
```
Comment (wwang-talend) [inline pom.xml:206]: will use next version, here only for local test, have merge that : https://github.com/jdereg/json-io/commit/3fea13ab41fe63aa2d4df1641b9d94c7979f8399

File: daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java [hunk 2]
```diff
@@ -34,8 +35,9 @@ public void testToBigDecimalEU() throws Exception {
 
     @Test
     public void testToBigDecimalGroupingEU() throws Exception {
-        assertEquals("4" + ((char) 160) + "512,5",
-                BigDecimalFormatter.format(new BigDecimal("4512.50"), BigDecimalParser.EU_DECIMAL_PATTERN));
+        // see https://kirillbelyaev.com/s/
```
Comment (sgandon) [inline daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: I don't understand why you have changed this test ?
The space is the grouping separator and this has not changed.
Please have a

File: daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java [hunk 3]
```diff
@@ -91,33 +92,23 @@ protected static void registerMavenUrlFactory() {
         // If the URL above failed, the mvn protocol needs to be installed.
         // not advice create a wrap URLStreamHandlerFactory class now
         try {
-            final Field factoryField = URL.class.getDeclaredField("factory");
```
Comment (sgandon) [inline daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java:94]: Why did you remove this check ?

File: daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java (hunk 2 above)
Comment (wwang-talend) [reply daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: if run it on java 17 runtime, the junit will fail as the "space" char is not "160", but another one.

File: daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java (hunk 3 above)
Comment (wwang-talend) [reply daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java:94]: since java 17, default don't allow call reflect on java package to access the private info, so if not remove this, will throw exception on java 17 here. And check the code, seems this is an old usage, no need to do that now.

File: daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java (hunk 2 above)
Comment (coheigea) [reply daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: @wwang-talend Is it possible instead of 160 to refer to a constant that will work with both JDK versions?

File: daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java (hunk 3 above)
Comment (sgandon) [reply daikon/src/main/java/org/talend/daikon/runtime/RuntimeUtil.java:94]: ok

File: daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java (hunk 2 above)
Comment (sgandon) [reply daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: @wwang-talend any answer on Colm proposal ?

Comment (wwang-talend) [reply daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: i will check it

Comment (wwang-talend) [reply daikon/src/test/java/org/talend/daikon/number/BigDecimalFormatterTest.java:38]: and now the regex pattern assert also works for both java17/11/8

**Review by sgandon — APPROVED** [review]
LGTM

**Review by sgandon — DISMISSED** [review]
LGTM

**Review by sgandon — APPROVED** [review]
LGTM
